control/control_test.go

```diff
@@ -123,117 +123,117 @@ func TestPluginControlGenerateArgs(t *testing.T) {
 	})
 }
 
-func TestSwapPlugin(t *testing.T) {
-	if SnapPath != "" {
-		c := New(GetDefaultConfig())
-		c.Start()
-		time.Sleep(100 * time.Millisecond)
-		lpe := newListenToPluginEvent()
-		c.eventManager.RegisterHandler("Control.PluginsSwapped", lpe)
-
-		_, e := load(c, PluginPath)
-		Convey("Loading first plugin", t, func() {
-			Convey("Should not error", func() {
-				So(e, ShouldBeNil)
-			})
-		})
-		// Travis optimization: If for some reason we can not load
-		// the plugin three times, we will fail the test immediately
-		// as we wait on a channel to be closed before proceeding with
-		// additional tests. If the plugin never loads, the channel will not
-		// close and just hang the test indefinitely.
-		if e != nil {
-			t.FailNow()
-		}
-		<-lpe.done
-		Convey("First plugin in catalog", t, func() {
-			Convey("Should have name mock", func() {
-				So(c.PluginCatalog()[0].Name(), ShouldEqual, "mock")
-			})
-		})
-		mock1Path := strings.Replace(PluginPath, "snap-collector-mock2", "snap-collector-mock1", 1)
-		mockRP, mErr := core.NewRequestedPlugin(mock1Path)
-		Convey("Loading a plugin should not error", t, func() {
-			So(mErr, ShouldBeNil)
-		})
-		err := c.SwapPlugins(mockRP, c.PluginCatalog()[0])
-		Convey("Swapping plugins", t, func() {
-			Convey("Should not error", func() {
-				So(err, ShouldBeNil)
-			})
-		})
-		if err != nil {
-			t.FailNow()
-		}
-		<-lpe.done
-
-		// Swap plugin that was loaded with a different version of the plugin
-		Convey("Swapping plugins", t, func() {
-			Convey("Should generate a swapped plugins event", func() {
-				Convey("So first plugin in catalog after swap should have name mock", func() {
-					So(c.PluginCatalog()[0].Name(), ShouldEqual, "mock")
-				})
-				Convey("So swapped plugins event should show loaded plugin name as mock", func() {
-					So(lpe.plugin.LoadedPluginName, ShouldEqual, "mock")
-				})
-				Convey("So swapped plugins event should show loaded plugin version as 1", func() {
-					So(lpe.plugin.LoadedPluginVersion, ShouldEqual, 1)
-				})
-				Convey("So swapped plugins event should show unloaded plugin name as mock", func() {
-					So(lpe.plugin.UnloadedPluginName, ShouldEqual, "mock")
-				})
-				Convey("So swapped plugins event should show unloaded plugin version as 2", func() {
-					So(lpe.plugin.UnloadedPluginVersion, ShouldEqual, 2)
-				})
-				Convey("So swapped plugins event should show plugin type as collector", func() {
-					So(lpe.plugin.PluginType, ShouldEqual, int(plugin.CollectorPluginType))
-				})
-			})
-		})
+// func TestSwapPlugin(t *testing.T) {
```

Comment:
This probably was commented out to save testing time or something, but should be uncommented.